fiona/ogrext.pyx

```diff
@@ -215,7 +205,351 @@ class TZ(datetime.tzinfo):
     def utcoffset(self, dt):
         return datetime.timedelta(minutes=self.minutes)
 
-# Feature extension classes and functions follow.
+
+cdef class AbstractField:
+
+    cdef object driver
+    cdef object supports_tz
+
+    def __init__(self, driver=None):
+        self.driver = driver
+
+    cdef object name(self, OGRFieldDefnH fdefn):
+        """Get the short Fiona field name corresponding to the OGR field definition."""
+        raise NotImplementedError
+
+    cdef object get(self, OGRFeatureH feature, int i, object kwds):
+        """Get the value of a feature's field."""
+        raise NotImplementedError
+
+    cdef set(self, OGRFeatureH feature, int i, object value, object kwds):
+        """Set the value of a feature's field."""
+        raise NotImplementedError
```

Comment:
The classes that derive from this one replace the ugly if/def statements in the feature builders.